### PR TITLE
Sscs 9872 record what types of documents get added or actioned

### PR DESCRIPTION
**Pull request by dattipoeHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-9872

### Change description ###
Add backend fields to track types of documents added to events.

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadService.java [hunk 1]
```diff
@@ -364,22 +371,30 @@ private void updateCaseDataWithNewAudioVideoUpload(List<SscsDocument> audioVideo
...
+                    .documentLink(sscsDocumentDetails.getDocumentLink())
+                    .dateAdded(ldt.toLocalDate())
+                    .fileName(sscsDocumentDetails.getDocumentFileName())
+                    .partyUploaded(uploader)
+                    .documentType(AudioVideoEvidenceUtil.getDocumentTypeValue(
+                        sscsDocumentDetails.getDocumentLink().getDocumentFilename()))
+                    .statementOfEvidencePdf(draftSscsDocument.getValue().getDocumentLink())
+                    .build())
+                .build());
+            System.out.println(sscsDocumentDetails.getDocumentLink().getDocumentFilename());
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadService.java:388]: Remove this System.out.println

Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadService.java:388]: Oops, debugger isn't working so having to use those

File: src/main/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadService.java [hunk 2]
```diff
@@ -67,29 +70,33 @@
 
     private static final String UPDATED_SSCS = "Updated SSCS";
     public static final String DM_STORE_USER_ID = "sscs";
-
+    private static final Enum<EventType> EVENT_TYPE = EventType.UPLOAD_DOCUMENT;
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadService.java:73]: Is this the right Event Type here?

Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadService.java:73]: The A/C call for that: Case [CCD_ID] with event uploadDocument added [documents] through MYA

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java [hunk 3]
```diff
@@ -331,39 +331,58 @@ private void buildSscsDocumentFromScan(SscsCaseData sscsCaseData, State caseStat
...
+        if (sscsCaseData.getSscsDocument() != null) {
+            documents.addAll(sscsCaseData.getSscsDocument());
+        }
 
-            if (REINSTATEMENT_REQUEST.getValue().equals(sscsDocument.getValue().getDocumentType())) {
-                if (isFurtherEvidenceActionCode(sscsCaseData.getFurtherEvidenceAction(), OTHER_DOCUMENT_MANUAL.getCode())) {
-                    setReinstateCaseFields(sscsCaseData);
-                }
-            }
+        sscsCaseData.setSscsDocument(documents);
```
Comment (chrisdavecm) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java:346]: Didn't this used to get executed for coversheets as well?

Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java:346]: Do you mean the documents being set? The coversheet was never added to the list of documents before: https://github.com/hmcts/sscs-tribunals-case-api/blob/master/src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java#L338-L366

**Review by chrisdavecm — APPROVED** [review]

**Pull request closed without merging**